code/modules/mob/living/whisper.dm

```diff
@@ -14,7 +14,7 @@
 	if(client.handle_spam_prevention(message, MUTE_IC))
 		return
 
-	whisper(message)
```

Comment:
Doing this means that whispering will no longer produce messages for eavesdroppers and watchers.
I think you need to find how it is avoiding the part that strips the language prefix in `/mob/living/say` or wherever.

Reply:
Tested it on a private server and I did not encounter the issues you mentioned, as the ordinary say part seems to take that into account as it runs through the code.